### PR TITLE
[shopsys] docs: added quick guide on how to see the progress of acceptance tests when using Docker

### PR DESCRIPTION
**Pull request by vitek-rostislav**
| Q             | A
| ------------- | ---
|Description, reason for the PR| We want our users to know this very handy information (on how to see the progress of acceptance tests when using Docker) that has been discovered by @PetrHeinz during implementation of https://github.com/shopsys/shopsys/pull/1245
|New feature| No 
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| No

### CODE REVIEW

File: docs/introduction/running-acceptance-tests.md [hunk 1]
```diff
@@ -22,6 +22,26 @@ php phing tests-acceptance
 *Note: In this step you were using Phing target `tests-acceptance`.
 More information about what Phing targets are and how they work can be found in [Console Commands for Application Management (Phing Targets)](/docs/introduction/console-commands-for-application-management-phing-targets.md)*
 
+### How to watch what is going on in the selenium browser
+By default, Shopsys Framework uses `selenium/standalone-chrome` image for `selenium-server` service which means you are not able to watch what is going on in the selenium browser.
+However, there is a quick solution which allows you to watch the progress of your acceptance tests:
+1. In your `docker-compose.yml`, use `standalone-chrome-debug` image for `selenium-server` service and new settings of ports:
+    ```diff
+    selenium-server:
```
Comment (PetrHeinz) [inline docs/introduction/running-acceptance-tests.md]:
```suggestion
      selenium-server:
```

File: docs/introduction/running-acceptance-tests.md [hunk 2]
```diff
@@ -22,6 +22,26 @@ php phing tests-acceptance
...
+    -    image: selenium/standalone-chrome:3.11
+    +    image: selenium/standalone-chrome-debug:3.11
+        container_name: shopsys-framework-acceptance-tests
+            ports:
+                - "4400:4444"
```
Comment (PetrHeinz) [inline docs/introduction/running-acceptance-tests.md]:
```suggestion
                  - "4400:4444"
```

File: docs/introduction/running-acceptance-tests.md [hunk 3]
```diff
@@ -22,6 +22,26 @@ php phing tests-acceptance
...
+### How to watch what is going on in the selenium browser
+By default, Shopsys Framework uses `selenium/standalone-chrome` image for `selenium-server` service which means you are not able to watch what is going on in the selenium browser.
+However, there is a quick solution which allows you to watch the progress of your acceptance tests:
+1. In your `docker-compose.yml`, use `standalone-chrome-debug` image for `selenium-server` service and new settings of ports:
+    ```diff
+    selenium-server:
+    -    image: selenium/standalone-chrome:3.11
+    +    image: selenium/standalone-chrome-debug:3.11
+        container_name: shopsys-framework-acceptance-tests
+            ports:
```
Comment (PetrHeinz) [inline docs/introduction/running-acceptance-tests.md]:
```suggestion
              ports:
```

File: docs/introduction/running-acceptance-tests.md [hunk 4]
```diff
@@ -22,6 +22,26 @@ php phing tests-acceptance
...
 
+### How to watch what is going on in the selenium browser
+By default, Shopsys Framework uses `selenium/standalone-chrome` image for `selenium-server` service which means you are not able to watch what is going on in the selenium browser.
+However, there is a quick solution which allows you to watch the progress of your acceptance tests:
+1. In your `docker-compose.yml`, use `standalone-chrome-debug` image for `selenium-server` service and new settings of ports:
+    ```diff
+    selenium-server:
+    -    image: selenium/standalone-chrome:3.11
+    +    image: selenium/standalone-chrome-debug:3.11
+        container_name: shopsys-framework-acceptance-tests
```
Comment (PetrHeinz) [inline docs/introduction/running-acceptance-tests.md]:
```suggestion
          container_name: shopsys-framework-acceptance-tests
```

File: docs/introduction/faq-and-common-issues.md [hunk 5]
```diff
@@ -125,4 +126,7 @@ Otherwise, the logic is in facades (resp. the facades are used as delegates to o
 See [Friendly URL](/docs/introduction/friendly-url.md) article.
 
 ## How can I create Front-end Breadcrumb navigation?
-See [Front-end Breadcrumb Navigation](/docs/introduction/front-end-breadcrumb-navigation.md) article.
\ No newline at end of file
+See [Front-end Breadcrumb Navigation](/docs/introduction/front-end-breadcrumb-navigation.md) article.
+
+## Is there any way how to see the progress of Codeception acceptance tests when using Docker?
```
Comment (grossmannmartin) [inline docs/introduction/faq-and-common-issues.md]: "... progress of Codeception acceptance tests ..." at first I thought about some progress bars while running.
I don't know better phrasing, but currently, it's not propagating the awesomeness enough 😄  
(not a CR just comment)

Comment (vitek-rostislav) [reply docs/introduction/faq-and-common-issues.md]: it took me quite a long time to come up with these words :laughing: another idea was something like this
```
Is there any way how to see what is going on in the Codeception acceptance tests when using Docker?
```

Comment (grossmannmartin) [reply docs/introduction/faq-and-common-issues.md]: What about

`Can I see what is really happening in the Codeception acceptance tests when using Docker?`

Comment (vitek-rostislav) [reply docs/introduction/faq-and-common-issues.md]: rewritten :wink:

File: docs/introduction/running-acceptance-tests.md [hunk 6]
```diff
@@ -22,6 +22,26 @@ php phing tests-acceptance
...
+    +    image: selenium/standalone-chrome-debug:3.11
+          container_name: shopsys-framework-acceptance-tests
+              ports:
+                  - "4400:4444"
+    +            - "5900:5900"
+    ```
+
+1. Run `docker-compose up -d`
+1. From your local machine, connect to the remote desktop on `vnc://127.0.0.1:5900`
+    - for the connection, you can use e.g. *Remmina* tool that is installed by default in Ubuntu
```
Comment (TomasLudvik) [inline docs/introduction/running-acceptance-tests.md]: On Mac you can run `open vnc://127.0.0.1:5900` in your terminal with password `secret`.

File: docs/introduction/running-acceptance-tests.md [hunk 7]
```diff
@@ -22,6 +22,26 @@ php phing tests-acceptance
...
 More information about what Phing targets are and how they work can be found in [Console Commands for Application Management (Phing Targets)](/docs/introduction/console-commands-for-application-management-phing-targets.md)*
 
+### How to watch what is going on in the selenium browser
+By default, Shopsys Framework uses `selenium/standalone-chrome` image for `selenium-server` service which means you are not able to watch what is going on in the selenium browser.
+However, there is a quick solution which allows you to watch the progress of your acceptance tests:
+1. In your `docker-compose.yml`, use `standalone-chrome-debug` image for `selenium-server` service and new settings of ports:
+    ```diff
+      selenium-server:
+    -    image: selenium/standalone-chrome:3.11
+    +    image: selenium/standalone-chrome-debug:3.11
```
Comment (TomasLudvik) [inline docs/introduction/running-acceptance-tests.md]: Would it not be easier to use this image by default?

Comment (TomasLudvik) [reply docs/introduction/running-acceptance-tests.md]: On line 8 in this document there is running sh instead of bash, you can update it

**Comment by vitek-rostislav** [comment]
Hi @TomasLudvik  - I have exchange `standalone-chrome` with `standalone-chrome-debug` in our `docker-compose files` :wink:

I will modify the commit message to take the fact into account after the fixup is approved :slightly_smiling_face:

File: docs/introduction/running-acceptance-tests.md (hunk 7 above)
Comment (vitek-rostislav) [reply docs/introduction/running-acceptance-tests.md]: fixed in https://github.com/shopsys/shopsys/pull/1251/commits/d5643221383bfaf33eacaf4c490e28646df23841

**Comment by TomasLudvik** [comment]
Hello @vitek-rostislav,

you are sugges

File: docs/introduction/running-acceptance-tests.md [hunk 8]
```diff
@@ -22,6 +22,13 @@ php phing tests-acceptance
 *Note: In this step you were using Phing target `tests-acceptance`.
 More information about what Phing targets are and how they work can be found in [Console Commands for Application Management (Phing Targets)](/docs/introduction/console-commands-for-application-management-phing-targets.md)*
 
+### How to watch what is going on in the selenium browser
+1. From your local machine, connect to the remote desktop on `vnc://127.0.0.1:5900`
```
Comment (TomasLudvik) [inline docs/introduction/running-acceptance-tests.md]:
```suggestion
1. From your local machine, connect to the remote desktop on `vnc://127.0.0.1:4400`
```

File: docs/introduction/running-acceptance-tests.md [hunk 9]
```diff
@@ -22,6 +22,13 @@ php phing tests-acceptance
...
+    - for the connection, you can use e.g. *Remmina* tool that is installed by default in Ubuntu
+    - on Mac, you can run `open vnc://127.0.0.1:5900` in your terminal
```
Comment (TomasLudvik) [inline docs/introduction/running-acceptance-tests.md]:
```suggestion
    - on Mac, you can run `open vnc://127.0.0.1:4400` in your terminal
```

**Comment by PetrHeinz** [comment]
`4400` is the port used by Selenium server. `5900` is the port of the remote desktop server, which is not exposed on the host machine anymore. 

In the original FAQ tip in https://github.com/shopsys/shopsys/commit/243ebc420aafbdb88d8855df5242aba622e9e29d#diff-398da4ce370335ce3aebecfcff346077, it was mentioned. Now, only Docker image has been switched. And the actual change was hidden by the fact the commits were rebased in the process.

File: docs/introduction/running-acceptance-tests.md (hunk 8 above)
Comment (PetrHeinz) [reply docs/introduction/running-acceptance-tests.md]: That's not the right port, doesn't connect to a remote desktop, see https://github.com/shopsys/shopsys/pull/1251#issuecomment-517697351

image

File: docs/upgrade/UPGRADE-unreleased.md [hunk 10]
```diff
@@ -4,3 +4,12 @@ This guide contains instructions to upgrade from version v8.0.0 to Unreleased.
...
 **Before you start, don't forget to take a look at [general instructions](/UPGRADE.md) about upgrading.**
 There you can find links to upgrade notes for other versions too.
+
+## Infrastructure
+- update your `docker-compose.yml` to use different image for `selenium-server` service ([#1251](https://github.com/shopsys/shopsys/pull/1251))
+    ```diff
+    selenium-server:
+    -    selenium/standalone-chrome:3.11
+    +    image: selenium/standalone-chrome-debug:3.11
+    ```
```
Comment (PetrHeinz) [inline docs/upgrade/UPGRADE-unreleased.md]:
```suggestion
    ```
    - now you can [watch what's going on in the Selenium browser](/docs/introduction/running-acceptance-tests.md#how-to-watch-what-is-going-on-in-the-selenium-browser) during the execution of acceptance tests
```

File: docs/introduction/running-acceptance-tests.md (hunk 8 above)
Comment (TomasLudvik) [reply docs/introduction/running-acceptance-tests.md]: Oh, ok there needs to be that port set in docker-compose

File: docs/introduction/running-acceptance-tests.md (hunk 9 above)
Comment (TomasLudvik) [reply docs/introduction/running-acceptance-tests.md]: Oh, ok there needs to be that port set in docker-compose

**Comment by vitek-rostislav** [comment]
there is some problem on Travis, however, the build is passing on Jenkins :slightly_smiling_face:

**Pull request merged**